### DISCUSSION TITLE
Open-Source vs. Cloud Version of InfraNodus

### DISCUSSION THREAD

**Discussion by deemeetree** [noduslabs/infranodus#331] [Announcements]
Just to let you know that we have the new Cloud version of InfraNodus available on [www.infranodus.com](https://infranodus.com)

A few years ago we had to pivot toward developing a scaleable product that could process bigger amounts of data and, unfortunately, it was not possible using the open-source infrastructure at the time. That's why the new InfraNodus has more capabilities and it also runs on PostgreSQL instead of Neo4J, using the more advanced frameworks, such as Nest.Js and Prisma. 

We decided to keep the open-source version, along with the installation instructions, available to everyone. However, we cannot support the codebase anymore. This version can still perform all the basic graph operations and network analysis stuff, so you can use it for some of the tasks. However, if you need to process bigger files, you will like the new Cloud version better. 

In the future, we plan to make API available and to create a Python-based open-source port for a part of the algorithm used by InfraNodus, to make a fully-capable version available to the open-source community and researchers.

**Comment by huyz** [comment]
Looking forward to seeing the "Python-based open-source port for a part of the algorithm" 👍

**Reply by deemeetree** [reply]
Oh that already exists since 2018. Just check out the Readme page :)

**Reply by huyz** [reply]
Oh then why did you write in 2022:

> In the future, we plan to make API available and to create a Python-based open-source port for a part of the algorithm used by InfraNodus, to make a fully-capable version available to the open-source community and researchers.

**Reply by catalystK** [reply]
@huyz stop being rude-dude..

**Reply by huyz** [reply]
How is that being rude? I'm asking a clarifying question.

Maybe instead of insulting me, you could enlighten me and answer the question.

**Reply by TomLucidor** [reply]
Might as well get people to "rewrite it in Rust" soon, and then add Python layer if necessary... Yeah InfraNodus only pops up in peoples radar cus they used to make a good text embedding model, rather than this network tool.

> Important: this is the old version of InfraNodus last updated in 2020. The new version with upgraded algorithms and AI capabilities that also includes the [browser extension](https://infranodus.com/extension) and the [Obsidian graph view plugin](https://infranodus.com/obsidian-plugin) is available at https://infranodus.com/. We do not provide support for this open-source version.

And THAT is how you kill hope for a useful project, worth forking and adding MCP as a community project tho considering how coding agents became viable https://github.com/noduslabs/infranodus/commits/master/ https://github.com/noduslabs/infranodus/forks
